### PR TITLE
Fix command for manually running redis

### PR DESCRIPTION
**Pull request by mariasfiraiala**
Signed-off-by: Maria Sfiraiala <maria.sfiraiala@gmail.com>

### CODE REVIEW

**Review by razvand — CHANGES_REQUESTED** [review]
Hi, @mariasfiraiala . What is this useful for? I was able to run the Redis Unikraft application with the initial command.

Your version of the command is required only if the Redis application is built with 9pfs support when running:

```
make menuconfig
```

But 9pfs support is not required. Maybe there should be a fix in the documentation of [`lib-redis`](https://github.com/unikraft/lib-redis/blob/staging/README.md) to point out that 9pfs support is **not** required (unlike [`lib-nginx`](https://github.com/unikraft/lib-nginx/blob/staging/README.md) - referenced there).

**Comment by mariasfiraiala** [comment]
Alright then, closing this PR. I'll take a look into the documentation for `lib-redis`.

**Comment by razvand** [comment]
Hi, @mariasfiraiala . Based on our discussion, this PR is indeed useful. We require a `redis.conf` configuration file to be passed to Redis, and that needs to be done via a filesystem (such as `9pfs`).

Please update this commit with the `fs0/redis.conf` file.

This `redis.conf` file worked for me, maybe you can simplify it:

```
bind 172.44.0.2
port 6379
tcp-backlog 511
timeout 0
tcp-keepalive 300
```

**Comment by razvand** [comment]
Also add instructions on adding `9pfs` support with the proper tag (`rootfs` in the command above), similar to the [`lib-nginx` documentation](https://github.com/unikraft/lib-nginx#root-filesystem). You could also add this to the [`lib-redis` repository](https://github.com/unikraft/lib-redis) for consistency.

I think consistency is important, so I would aim to make `Redis` and `Nginx` as close to each other as possible, documentation-wise.

File: README.md [hunk 1]
```diff
@@ -90,18 +90,32 @@ To configure it for the KVM platform:
...
+Create a directory (`fs0`) used for mounting the file system. It should be placed inside the `app-redis` directory.
+
+Add a configuration file for `redis` (named `redis.conf`) in the directory used for mounting the file system:
+```
+bind 172.44.0.2
+port 6379
+tcp-backlog 511
+timeout 0
+tcp-keepalive 300
+```
```
Comment (StefanJum) [inline README.md]: This will al be done bt #10, there is no need to have it here.

**Review by StefanJum — COMMENTED** [review]
Only one comment, looks good besides that.

**Comment by mariasfiraiala** [comment]
> Only one comment, looks good besides that.

Thank you @StefanJum . Removed it.

File: README.md [hunk 2]
```diff
@@ -90,18 +90,21 @@ To configure it for the KVM platform:
...
 Build the application:
 ```
 $ make
 ```
 
 Run the application:
 ```
-sudo qemu-system-x86_64 \
+sudo qemu-system-x86_64 -fsdev local,id=myid,path=$(pwd)/fs0,security_model=none \
+                        -device virtio-9p-pci,fsdev=myid,mount_tag=rootfs,disable-modern=on,disable-legacy=off \
```
Comment (StefanJum) [inline README.md]:
```suggestion
                        -device virtio-9p-pci,fsdev=myid,mount_tag=fs0,disable-modern=on,disable-legacy=off \
```
If #10 is merged, `kraft` will use the default `fs0` mount tag.

File: README.md [hunk 3]
```diff
@@ -90,18 +90,21 @@ To configure it for the KVM platform:
 $ make menuconfig
 ```
 
+Be aware of the fact that you'll have to choose a file system: for instance, from the `vfscore` library, choose the `Default root filesystem` to be `9pfs`.
```
Comment (StefanJum) [inline README.md:93]: We should also specify that the mount tag must be `fs0` if the users will want to use `kraft` or `qemu-guest`.

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few more comments.

File: README.md (hunk 3 above)
Comment (razvand) [reply README.md:93]: Agree with @StefanJum 

@mariasfiraiala , I may fucked things up as I did a force push with some updates.

Please use the new updated branch, see @StefanJum 's comments, mark conversations solved by clicking `Resolve conversation` (once they are solved) and re-request his review.

File: README.md [hunk 4]
```diff
@@ -90,18 +90,23 @@ To configure it for the KVM platform:
...
+
+If you are going to use the `qemu` or kraft, wrapper to launch the app, you need to name the `Default root device` `fs0` (`Library Configuration` -> `vfscore` -> `Default root device`). This is due to how the `qemu-guest` script and krat automatically tag the FS devices attached to `qemu`.
```
Comment (StefanJum) [inline README.md]:
```suggestion
If you are going to use the `qemu-guest` script or `kraft` to launch the app, you need to name the `Default root device` `fs0` (`Library Configuration` -> `vfscore` -> `Default root device`).
This is due to how the `qemu-guest` script and krat automatically tag the FS devices attached to `qemu`.
```

Please use one line per sentence and one sentence per line.

**Review by StefanJum — CHANGES_REQUESTED** [review]
One more little comment.

File: README.md [hunk 5]
```diff
@@ -90,18 +90,24 @@ To configure it for the KVM platform:
 $ make menuconfig
 ```
 
+Be aware of the fact that you'll have to choose a file system: for instance, from the `vfscore` library, choose the `Default root filesystem` to be `9pfs`.
+
+If you are going to use the `qemu-guest` script or `kraft`, wrapper to launch the app, you need to name the `Default root device` `fs0` (`Library Configuration` -> `vfscore` -> `Default root device`).
+This is due to how the `qemu-guest` script and krat automatically tag the FS devices attached to `qemu`.
```
Comment (StefanJum) [inline README.md]:
```suggestion
This is due to how the `qemu-guest` script and kraft automatically tag the FS devices attached to `qemu`.
```

File: README.md [hunk 6]
```diff
@@ -90,18 +90,24 @@ To configure it for the KVM platform:
 $ make menuconfig
 ```
 
+Be aware of the fact that you'll have to choose a file system: for instance, from the `vfscore` library, choose the `Default root filesystem` to be `9pfs`.
+
+If you are going to use the `qemu-guest` script or `kraft`, wrapper to launch the app, you need to name the `Default root device` `fs0` (`Library Configuration` -> `vfscore` -> `Default root device`).
```
Comment (StefanJum) [inline README.md]:
```suggestion
If you are going to use the `qemu-guest` script or `kraft` to launch the app, you need to name the `Default root device` `fs0` (`Library Configuration` -> `vfscore` -> `Default root device`).
```

**Review by StefanJum — APPROVED** [review]
All good, thank you @mariasfiraiala.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Comment by razvand** [comment]
@mariasfiraiala , please mark solved items with `Resolve conversation` and re-request reviews from reviewers (@StefanJum in this case).

**Review by razvand — APPROVED** [review]
This works. Thanks, @mariasfiraiala 

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**